File: [24.01]Spring-Batch/Java/src/main/java/com/example/java/batch/TaskLetJobConfig1.java
package com.example.java.batch;

import lombok.extern.slf4j.Slf4j;
import org.springframework.batch.core.Job;
import org.springframework.batch.core.Step;
import org.springframework.batch.core.job.builder.JobBuilder;
import org.springframework.batch.core.repository.JobRepository;
import org.springframework.batch.core.step.builder.StepBuilder;
import org.springframework.batch.core.step.tasklet.Tasklet;
import org.springframework.batch.repeat.RepeatStatus;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.transaction.PlatformTransactionManager;

// 람다 형식
@Slf4j
@Configuration
public class TaskLetJobConfig1 {

    @Bean
    public Step taskStep(JobRepository jobRepository, PlatformTransactionManager platformTransactionManager) {
        log.info(">>>> SingleStep");
        return new StepBuilder("taskStep", jobRepository).tasklet(((contribution, chunkContext) -> {
            for(int i =0; i<10; i++){
                log.info(i+": 비즈니스 로직");
            }
            return RepeatStatus.FINISHED;
        }), platformTransactionManager).build();
    }

    @Bean
    public Job taskJob(JobRepository jobRepository, Step taskStep) {
        log.info(">>>> SingleJob");
        return new JobBuilder("taskJob", jobRepository)
                .start(taskStep)
                .build();
    }
}

File: [24.01]Spring-Batch/kotlin/src/main/kotlin/com/example/kotlin/tasklet/TaskletJobConfig1.kt
package com.example.kotlin.tasklet

import org.slf4j.LoggerFactory
import org.springframework.batch.core.Job
import org.springframework.batch.core.Step
import org.springframework.batch.core.job.builder.JobBuilder
import org.springframework.batch.core.repository.JobRepository
import org.springframework.batch.core.step.builder.StepBuilder
import org.springframework.batch.repeat.RepeatStatus
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.transaction.PlatformTransactionManager

@Configuration
class TaskletJobConfig1 {
    private val log = LoggerFactory.getLogger(this.javaClass)!!

    @Bean
    fun taskletStep(
        jobRepository: JobRepository,
        platformTransactionManager: PlatformTransactionManager
    ): Step {
        log.info(">>>> taskletStep1")
        return StepBuilder("flowStep1", jobRepository).tasklet({ contribution, chunkContext ->
            for (i in 0..10) {
                log.info("람다식 ${i}번째 비지니스 로직");
            }
            RepeatStatus.FINISHED
        }, platformTransactionManager).build()
    }

    @Bean
    fun taskletJob(jobRepository: JobRepository, taskletStep: Step): Job {
        log.info(">>>> taskletJob1")
        return JobBuilder("taskletJob1", jobRepository)
            .start(taskletStep)
            .build()
    }

}